### PR TITLE
wallet-ext: BackgroundServiceSigner sign data signature

### PR DESCRIPTION
**Pull request by pchrysochoidis**
## Description 
* update BackgroundServiceSigner to create a signature by signing the data digest instead the data itself
* make sure BackgroundServiceSigner produces the same signature as RawSigner

## Test Plan 
pnpm test
also try sending sui from wallet

fixes executing transactions from wallet (due to wrong signature) [after changing the way signature is created](https://github.com/MystenLabs/sui/pull/9561)

### CODE REVIEW

**Review by kchalkias — APPROVED** [review]

File: apps/wallet/src/background/keyring/AccountKeypair.ts [hunk 1]
```diff
@@ -15,9 +16,10 @@ export class AccountKeypair {
     }
 
     async sign(data: Uint8Array): Promise<SerializedSignature> {
+        const digest = blake2b(data, { dkLen: 32 });
```
Comment (williamrobertson13) [inline apps/wallet/src/background/keyring/AccountKeypair.ts:19]: why `32`?

**Review by williamrobertson13 — APPROVED** [review]

File: apps/wallet/src/background/keyring/AccountKeypair.test.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import { describe, expect, it } from 'vitest';
+
+import { AccountKeypair } from './AccountKeypair';
+
+describe('AccountKeypair', () => {
+    describe('signData produces same signature with RawSigner', () => {
+        it.each([
+            { label: 'Ed25519Keypair', keypair: new Ed25519Keypair() },
+            { label: 'Secp256k1Keypair', keypair: new Secp256k1Keypair() },
+        ])('for $label', async ({ label, keypair }) => {
```
Comment (williamrobertson13) [inline apps/wallet/src/background/keyring/AccountKeypair.test.ts:19]: does this need to be a template string? I haven't written too many `it.each` tests

File: apps/wallet/src/background/keyring/AccountKeypair.ts (hunk 1 above)
Comment (kchalkias) [reply apps/wallet/src/background/keyring/AccountKeypair.ts:19]: @williamrobertson13 good question, we use blake2b with 32byte (256bits) outputs

Comment (williamrobertson13) [reply apps/wallet/src/background/keyring/AccountKeypair.ts:19]: thanks for the explainer! not required for this diff, but I wonder if we should codify this better at the SDK level with a constant, e.g. `TRANSACTION_DIGEST_LENGTH`

**Pull request merged**